File: ProcedureStyle/regtet.cpp
// triangle.cpp - содержит функции обработки правильного тэтраэдра

#include "regtet.h"
#define SQRT3 1.73205081

// Ввод параметров правильного тэтраэдра из потока
void in(regtet& rt, ifstream& ifst) {
    ifst >> rt.edge;
}

// Случайный ввод параметров правильного тэтраэдра
void in_rand(regtet& rt) {
    rt.edge = Random();
}

// Вывод параметров правильного тэтраэдра в поток
void out(regtet& rt, ofstream& ofst) {
    ofst << "It is Regular Tetrahedron: edge = " << rt.edge << ". Surface square = " << surf_sq(rt) << endl;
}

// Вычисление площади поверхности правильного тэтраэдра
double surf_sq(regtet& rt) {
    return double(rt.edge * rt.edge * SQRT3);
}

File: DynamicTyping/regtet.py
# triangle.ру содержит функции обработки правильного тэтраэдра
from random import randrange
from shape import *
from math import sqrt

class regtet(shape):
    # Инициализация правильного тэтраэдра
    def __init__(self):
        self.edge = 0

    # Рандомная генерация правильного тэтраэдра
    def in_rand(self):
        self.edge = randrange(50) + 1
        pass

    # Чтение правильного тэтраэдра из файла
    def read_string_arr(self, str_arr, i):
        if i >= len(str_arr):
            return 0
        self.edge = int(str_arr[i])
        i += 1
        return i

    # Вывод правильного тэтраэдра в консоль
    def print(self):
        print("Regular tetrahedron: edge = ", self.edge, ". Surface square = ", self.surf_sq())
        pass

    # Вывод правильного тэтраэдра в поток (для записи в файл)
    def write(self, ostream):
        ostream.write("Regular tetrahedron: edge = {}. Surface square = {}".format(self.edge, self.surf_sq()))
        pass

    # Поиск площади поверхности правильного тэтраэдра
    def surf_sq(self):
        return float((self.edge ** 2) * sqrt(3))
        pass
